satellite/admin/project.go

```diff
@@ -285,3 +285,40 @@ func (server *Server) addProject(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	_, _ = w.Write(data) // nothing to do with the error response, probably the client requesting disappeared
 }
+
+func (server *Server) deleteProject(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	vars := mux.Vars(r)
+	projectUUIDString, ok := vars["project"]
+	if !ok {
+		http.Error(w, "project-uuid missing", http.StatusBadRequest)
+		return
+	}
+
+	projectUUID, err := uuid.FromString(projectUUIDString)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("invalid project-uuid: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	if err := r.ParseForm(); err != nil {
+		http.Error(w, fmt.Sprintf("invalid form: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	keys, err := server.db.Console().APIKeys().GetPagedByProjectID(ctx, projectUUID, console.APIKeyCursor{Limit: 2, Page: 1})
+	if err != nil {
+		http.Error(w, fmt.Sprintf("unable to list api-keys: %v", err), http.StatusInternalServerError)
+		return
+	}
+	if keys.TotalCount > 0 {
+		http.Error(w, fmt.Sprintf("api-keys still exist"), http.StatusConflict)
+	}
+
+	err = server.db.Console().Projects().Delete(ctx, projectUUID)
```

Comment:
I believe you should also check there are no buckets left before deleting the project.

Reply:
Added the check